### ISSUE TITLE
hexdump context by bytes granularity

### ISSUE THREAD

**Issue by tonybounty** [Genivia/ugrep#171]
Is there a way to display matches in hexdump with specified X bytes before and Y bytes after ?

For example, in this screenshot I would like to display 5 bytes before and 5 bytes after the string "WinSCP" :

![image](https://user-images.githubusercontent.com/47276302/143771995-83f74830-d48a-441b-8a41-d63f4c24da9a.png)

**Comment by genivia-inc** [comment]
Yes, you can do this with option `-y` to show all context, or just add some dots before and/or after the pattern like `..(pattern)..`. The dots will match anything except newlines (use `--dotall` to match anything). One small caveat is that the dots will be shown as highlighted matches, not highlighted as context.

**Comment by genivia-inc** [comment]
Nothing wrong with your question, although the method to add "dots" before and after a pattern has been recommended by the community before, see e.g. https://stackoverflow.com/questions/8101701/grep-characters-before-and-after-match

In your case od this: `.{0,5}(pattern).{0,5}`.

**Comment by genivia-inc** [comment]
I'm revisiting this question, because I think there is a nice solution for this.

What if we show a number of hex lines before and/or after the match? Like this:

![image](https://user-images.githubusercontent.com/13117347/152695643-83de40b2-196e-46c4-b05d-1472bb6b3deb.png)

The `--hexdump` option will offer the option to show the specified number of hex lines before and after the matches:
~~~
    --hexdump=[1-8][a][b][c][h][A[NUM]][B[NUM]][C[NUM]]
            Output matches in 1 to 8 columns of 8 hexadecimal octets.  The
            default is 2 columns or 16 octets per line.  Option `a' outputs a
            `*' for all hex lines that are identical to the previous hex line,
            `b' removes all space breaks, `c' removes the character column, `h'
            removes hex spacing, `A' includes up to NUM hex lines after the
            match, `B' includes up to NUM hex lines before the match and `C'
            includes up to NUM hex lines.  When NUM is omitted, the matching
            line is included.  See also options -U, -W and -X.
~~~
For example, `--hexdump=C1` shows up to one extra hex line before and after the match, `--hexdump` shows no extra hex lines (same as `--hexdump=C0`), and `--hexdump=C` shows the matching line as hex lines of byte context, which is the same as `-X`. Also `--hexdump=B1A2` works with one line before and one line after.

I like the fact that `--hexdump` shows a simplified output without the entire matching line in hex, which can be very long. This differs from `-X`, which will have the old behavior to display the entire matching line in hex.

Internally ugrep still sees line breaks. So if there is a line break it will respect the break in favor of the NUM arguments of the `--hexdump` options `A`, `B` and `C`. It will not include more before/after hex bytes if those NUM values exceed the line break points. This is to ensure that the other usual context options `-A`, `-B`, `-C` also work.

This is a workable approach without breaking existing options, or perhaps there is an even better way of accomplishing something like this. Suggestions welcome!
